src/source/CppMarshal.scala

```diff
@@ -6,27 +6,38 @@ import djinni.meta._
 
 class CppMarshal(spec: Spec) extends Marshal(spec) {
 
```

Comment:
The purpose of the scopeNames variable used pervasively here doesn't seem likely to be clear outside the context of this pull request.  It could use a comment somewhere documenting its purpose.  Also maybe a more explicit name like conflictingSymbolsInScope?